.elasticbeanstalk/package.sh

```diff
@@ -1,23 +1,44 @@
 #!/usr/bin/env bash
 
-# Helper for packaging the app for elasticbeanstalk deployment.
+# Packages and uploads the app for elasticbeanstalk deployment
 #
-# Expected environment variables (should be set set by semaphore)
-#   - BRANCH_NAME
-#   - SEMAPHORE_BUILD_NUMBER
+# Expected environment variables (or circleci equivalents)
+#   Provided by caller
+#     - AWS_DEFAULT_REGION
+#     - S3_BUCKET_NAME
+#     - EB_APP_NAME
+#
+#   Provided by CI
+#     - BRANCH_NAME
+#     - BUILD_NUMBER
 #
 # Usage: .elasticbeanstalk/package.sh APP
 # Ex: .elasticbeanstalk/package.sh ${SEMAPHORE_PROJECT_NAME}
 
-set -euo pipefail
+set -euxo pipefail
+
+BRANCH_NAME="${CIRCLE_BRANCH:-${BRANCH_NAME}}"
+BUILD_NUMBER="${CIRCLE_BUILD_NUM:-${BUILD_NUMBER}}"
+DESCRIPTION="${CIRCLE_BUILD_URL:-${DESCRIPTION}}"
 
-BRANCH_NAME="${BRANCH_NAME:-${SEMAPHORE_GIT_BRANCH}}"
-SEMAPHORE_BUILD_NUMBER="${SEMAPHORE_BUILD_NUMBER:-${SEMAPHORE_WORKFLOW_ID}}"
+CLEAN_BRANCH_NAME="${BRANCH_NAME//\//_}"
+CLEAN_DESCRIPTION="${DESCRIPTION/https*:\/\//}"
```

Comment:
Just curious about why these characters were causing issues with the AWS CLI (I assume). I'm guessing the description purge is related to the CLI automatically trying to GET URLs like you were mentioning the other week?

Reply:
exactly. The branch name is used as a file name (e.g., `api/api-pull_565-549.zip`) where slashes would introduce a subpath.

And the http:// in any aws cli field causes it to try to insert the body of the http response as the arg unless you configure it off first. Rather than requiring anyone running this to first do an aws cli configuration it seemed best to drop the url schema which stops the behavior.

And the result on the beanstalk UI is still pretty good I think (it won't auto-link descriptions with or without the http).

<img width="892" alt="Screen Shot 2019-07-03 at 4 21 05 PM" src="https://user-images.githubusercontent.com/690/60571427-93242280-9dae-11e9-879b-5d7a895b01d7.png">